### PR TITLE
(asm) Use kpt to set asm namepsace label, use revision to download install_asm script and package. Issue #144, Fix #224 

### LINKED ISSUE #224: Need to use kpt to update all namespace labels for specific ASM version.

### PR DESCRIPTION
**Pull request by zijianjoy**
Issue #144
Fix #224 

Document how to pin ASM version in Makefile.
Use kpt cfg set for all ASM namespace labels. The value of these labels `istio.io/rev=asm-XXX-X` need to be consistent across cluster.

### CODE REVIEW

File: kubeflow/common/knative/Kptfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+metadata:
+  name: profiles
+packageMetadata:
+  shortDescription: profile controller config.
```
Comment (Bobgy) [inline kubeflow/common/knative/Kptfile:6]: Nit, update for knative

File: kubeflow/common/kubeflow-namespace/Kptfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+metadata:
+  name: profiles
+packageMetadata:
+  shortDescription: profile controller config.
```
Comment (Bobgy) [inline kubeflow/common/kubeflow-namespace/Kptfile:6]: Nit: update package name and description

File: kubeflow/kpt-set.sh [hunk 3]
```diff
@@ -2,6 +2,7 @@
 # Use kpt to set kustomize values
 
 kpt cfg set -R .  gke.private false
+kpt cfg set -R .  asm-label asm-192-1
```
Comment (Bobgy) [inline kubeflow/kpt-set.sh:5]: Make this a env var?

**Review by Bobgy — COMMENTED** [review]
LGTM, some minor comments

File: kubeflow/kpt-set.sh (hunk 3 above)
Comment (zijianjoy) [reply kubeflow/kpt-set.sh]: Sounds good, I changed it to env var.

File: kubeflow/common/kubeflow-namespace/Kptfile (hunk 2 above)
Comment (zijianjoy) [reply kubeflow/common/kubeflow-namespace/Kptfile]: Thank you! Applied.

File: kubeflow/common/knative/Kptfile (hunk 1 above)
Comment (zijianjoy) [reply kubeflow/common/knative/Kptfile]: Thank you! Applied.

**Comment by Bobgy** [comment]
/lgtm
/approve

Awesome 👍

**Pull request merged**